docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,242 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "    # replace DB_CHARM with desired unit name: `katib-db` or `kfp-db`\n",
+    "    DB_CHARM= <kfp-db | katib-db>\n",
+    "    juju show-application $DB_CHARM | yq '.[] | .charm'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "- Enough storage in the cluster to support backup/restore of the databases.\n",
+    "- `mysql-client` on client machine (install by running `sudo apt install mysql-client`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Obtain existing database credentials\n",
+    "\n",
+    "To obtain database credential execute the following commands for each database to be migrated and use those credentials in migration steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# replace DB_UNIT with desired unit name: `katib-db/0` or `kfp-db/0`\n",
+    "DB_UNIT= <katib-db/0 | kfp-db/0>\n",
+    "\n",
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.7 username used is `root` and password is specified in `mysql` relation by\n",
+    "# 'root_password'\n",
+    "DB_MYSQL_RELATION_ID=$(juju show-unit $DB_UNIT | yq '.[] | .relation-info | select(.[].endpoint == \"mysql\") | .[0] | .relation-id')\n",
+    "DB_USER=root\n",
+    "DB_PASSWORD=$(bash -c \"juju run --unit $DB_UNIT 'relation-get -r $DB_MYSQL_RELATION_ID - $DB_UNIT' | grep root_password\" | awk '{print $2}')\n",
+    "DB_IP=$(juju show-unit $DB_UNIT/0 | yq '.[] | .address')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy MySQL databases\n",
+    "\n",
+    "Deploy new MySQL database and obtain credentials by executing the following commands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# replace MYSQL_DB with desired application name: `katib-db-mysql` or `kfp-db-mysql`\n",
+    "MYSQL_DB= <katib-db-mysql/0 | kfp-db-mysql/0>\n",
+    "\n",
+    "# deploy new MySQL database charm\n",
+    "juju deploy mysql-k8s $MYSQL_DB --channel 8.0/stable --trust\n",
+    "\n",
+    "# obtain username and password of new MySQL database from MySQL charm\n",
+    "MYSQL_DB_USER=$(juju run-action $MYSQL_DB get-password --wait | yq '.[] | .results.username')\n",
+    "MYSQL_DB_PASSWORD=$(juju run-action $MYSQL_DB get-password --wait | yq '.[] | .results.password')\n",
+    "MYSQL_DB_IP=$(juju show-unit $MYSQL_DB | yq '.[] | .address')\n"
```

Comment:
You have to be careful where exactly the unit number is used and when the application is used. For instance running `juju deploy mysql-k8s $MYSQL_DB --channel 8.0/stable --trust` after setting `MYSQL_DB=katib-db-mysql/0` will give you a name error.

Reply:
Updating to conver for this. Please see comment above about variables.